src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -393,6 +398,23 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         manager.register_pass<ov::pass::MarkDequantization>(
             std::vector<ov::element::Type>{ ov::element::i8, ov::element::u8, ov::element::i4, ov::element::u4 },
             !device_info.supports_immad);
+        const bool disable_moe_opt = GPU_DEBUG_VALUE_OR(config.get_disable_moe_opt(), false);
+        if (!disable_moe_opt) {
+            manager.register_pass<ov::pass::FuseVectorizedMOE2GEMM>();
+        }
+        pass_config->set_callback<ov::pass::FuseVectorizedMOE2GEMM>([&](const_node_ptr& root) -> bool {
+            auto& engine = m_context->get_engine();
+            const auto& info = engine.get_device_info();
+            return (info.supports_immad == false) && info.arch == cldnn::gpu_arch::xe2;
+        });
```

Comment:
* could you add comment about this condition? It is difficult to understand the condition.
* I guess it is support on >= xe2?
* I could not see `transformation_callback` call within FuseVectorizedMOE2GEMM pass. Is this callback executed still?

Reply:
oops I fixed. Thanks!